### PR TITLE
migrate app to appCompat, and migrate to androidx

### PR DESCRIPTION
**Pull request by TacoTheDank**
Signed-off-by: TacoTheDank <SkytkRSfan3895@gmail.com>

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -2,13 +2,15 @@ apply plugin: 'com.android.application'
 
 android {
     compileSdkVersion 28
+    buildToolsVersion '28.0.3'
```
Comment (luk1337) [inline app/build.gradle]: Do we really need to specify build tools version here?

File: .gitignore [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
-/local.properties
-/.idea
+.gradle/
+local.properties
+.idea/
 .DS_Store
-/build
-/captures
+build/
+captures/
```
Comment (luk1337) [inline .gitignore]: Is there any reason to do that?

**Comment by luk1337** [comment]
I have picked the changes manually, splitting them into few smaller commits.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <resources>
     <string name="app_name">SauceNAO</string>
 
-    <string name="select_image">Select image</string>
+    <string name="select_image">Select Image</string>
```
[the remark this answers is not in the document]
Comment (TacoTheDank) [reply app/src/main/res/values/strings.xml]: I guess why not do it?

File: app/build.gradle (hunk 1 above)
Comment (TacoTheDank) [reply app/build.gradle]: It's not necessary, but it's nice to have

File: .gitignore (hunk 2 above)
Comment (TacoTheDank) [reply .gitignore]: I've had some issues where .gitignore doesn't detect /build (but detects build/), so eh...

**Comment by TacoTheDank** [comment]
That's alright, thank you for your time :3

**Pull request closed without merging**